### DISCUSSION TITLE
What exactly is machine learning for?

### DISCUSSION THREAD

**Discussion by anthonyma94** [immich-app/immich#14405] [Q&A]
From what I can tell, Immich does not have regular text search, i.e., it can't search for "cat" for all photos of cats. If I'm correct, what exactly is the machine learning library for?

**Comment by danieldietzler** [comment]
You can do exactly that. Search for `cat` to see pictures of cats. That's what ML is used for

**Reply by anthonyma94** [reply]
Oh, I must have set up my server wrong then, my bad!

**Reply by danieldietzler** [reply]
Looking at the logs will probably help :)

**Reply by anthonyma94** [reply]
I just tried it again, typing in "cat" brings up one photo of a person (and no cat). I'll rerun the machine learning job and try again.

**Reply by danieldietzler** [reply]
ML orders by relevance and we currently don't have any cutoff. So you'll essentially see *every* image if you scroll down far enough. The best matching images will just be at the top

**Reply by PhobosK** [reply]
Any plans for implementing some kind of a cutoff?

BTW great app and support :)
Thanks for your work.

**Reply by bo0tzz** [reply]
It's nontrivial to do, so we have no specific plans.
